apps/balancer-tools/src/app/apr/(utils)/tokenYield.ts

```diff
@@ -20,43 +20,133 @@ export const publicClient = createPublicClient({
 });
 
 const SECONDS_IN_DAY = 86400;
+const DAYS_IN_YEAR = 365;
+const SECONDS_IN_YEAR = DAYS_IN_YEAR * SECONDS_IN_DAY;
 
-//TODO: remove this on BAL-797
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-export async function getTokenAprByPoolId(poolId: string) {
-  const { currentRate, rate24hAgo } = await getDailyRateDiff();
-
-  return [
-    {
-      address: "0xae78736cd615f374d3085123a210448e74fc6393",
-      symbol: "rETH",
-      yield: ((currentRate - rate24hAgo) / 10 ** 18) * 365 * 100,
-    },
-  ];
+export async function getPoolTokensAprForDate(
+  chain: string,
+  poolId: Address,
+  date: number,
+) {
+  const rateProviders = await getPoolTokensRateProviders(chain, poolId);
+
+  return await Promise.all(
```

Comment:
use pThrottle?

Reply:
why? have you seen this getting rate limited? we won't have many rates here most times

Reply:
yup, this is calling multiple times, but not sure why yet

Reply:
rn there are some pools that we have 3 tokens at we will do this request 3*number of existing rounds for this pool

Reply:
example of a pool I got rate limited: https://tools-monorepo-git-luiza-bal-784-validate-getrat-01ffd7-bleu-fi.vercel.app/apr/pool/ethereum/0x42ed016f826165c2e5976fe5bc3df540c5ad0af700000000000000000000058b/round/76

Reply:
local dev if I retry:
![image](https://github.com/bleu-studio/balancer-tools/assets/63298341/eaeeab9e-5a62-484f-a51e-6f3bd7430282)